src/main/java/edu/cornell/kfs/coa/businessobject/CuAccountGlobal.java

```diff
@@ -124,11 +124,40 @@ private Account updateAccountValuesOnAccountGlobalDetailWithNewValuesFromAccount
 			if(ObjectUtils.isNotNull(this.getIndirectCostRecoveryAccounts()) && this.getIndirectCostRecoveryAccounts().size() > 0){
 				updateIcrAccounts(globalDetail, account.getIndirectCostRecoveryAccounts());
 			}
+
+			if (ObjectUtils.isNotNull(this.accountRestrictedStatusCode)) {
+				updateAccountRestrictedStatusCodeForAccountBeingEdited(this.accountRestrictedStatusCode, account);
+			}
 		}
 
 		return account;
 	}
 
+	/**
+	 * The business rules for the accountRestrictedStatusCode is as follows:
+	 * If the sub-fund of the account listed in the edit list of accounts on
+	 * the global maintenance edoc is one that has a default accountRestrictedStatusCode
+	 * listed, then any change entered in the global account maintenance edoc is ignored.
+	 *
+	 * If the sub-fund of the account listed in the edit list of accounts on
+	 * the global maintenance edoc is one that does NOT have a default accountRestrictedStatusCode
+	 * listed, then any change entered in the global account maintenance edoc overwrites
+	 * the accountRestrictedStatusCode value on that particular account from the edit list.
+	 *
+	 * @param globalAccountMaintenanceDocRestrictedStatusCode
+	 * @param accountBeingEdited
+	 */
+	private void updateAccountRestrictedStatusCodeForAccountBeingEdited(String globalAccountMaintenanceDocRestrictedStatusCode, Account accountBeingEdited)
+	{
+		if (ObjectUtils.isNotNull(accountBeingEdited.getSubFundGroup()) && StringUtils.isNotBlank(accountBeingEdited.getSubFundGroup().getAccountRestrictedStatusCode())) {
+			//ignore the global account maintenance edoc user entered value because the sub-fund for the account already has a default value listed that should be used
+		}
+		else {
+			//sub-fund associated to the account does NOT have a default accountRestrictedStatusCode identified, utilize the user entered value from the  global account maintenance edoc
+			accountBeingEdited.setAccountRestrictedStatusCode(globalAccountMaintenanceDocRestrictedStatusCode);
```

Comment:
This will not work correctly if the Account global doc is changing the sub-fund group code. The restricted status code will be set from the existing sub fund group rather than updated one.